src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -1005,7 +1005,7 @@ public static Task WriteAllLinesAsync(string path, IEnumerable<string> contents,
                 int index = 0;
                 while (index < count)
```

Comment:
Also, `count` is typically used to mean "number of items being copied", e.g. when taking buffer/offset/count triplets.